util/archive_builder.go

```diff
@@ -164,18 +164,18 @@ func copyDirectory(src, dest string) error {
 
 		if fd.IsDir() {
 			if err = copyDirectory(srcfp, dstfp); err != nil {
-
+				return err
```

Comment:
It will be good to have unit test about this.

Reply:
There _are_ unit tests for `archive_handler` that handle errors when working with files

Reply:
I mean that after you start to return error, some test should be start to fail. If it's not probably there is no test for such cases.